packages/react-components/analytics/CeloAnalytics.tsx

```diff
@@ -116,53 +121,63 @@ class CeloAnalytics {
       _.set(props, 'device', this.deviceInfo)
     }
     Analytics.track(eventName, props).catch((err) => {
-      this.Logger.error(TAG, `Failed to tracking event ${eventName}`, err)
+      this.Logger.error(TAG, `Failed to track event ${eventName}`, err)
     })
   }
 
   // Used with trackSubEvent and endTracking to track durations for
   // processes with multiple steps. For one-off events, use track method
-  startTracking(eventName: string) {
+  // Duplicate event properties will reflect latest value provided
+  startTracking(eventName: string, eventProperties = {}) {
     this.activeEvents.set(
       eventName,
-      new Map<string, number>([['__startTrackingTime__', Date.now()]])
+      new Map<string, SubEventData>([
+        ['__startTracking__', { timestamp: Date.now(), subEventProps: eventProperties }],
+      ])
     )
   }
 
   // See startTracking
-  trackSubEvent(eventName: string, subEventName: string) {
+  trackSubEvent(eventName: string, subEventName: string, eventProperties = {}) {
```

Comment:
should eventProperties be typed here to only allow certain keys?

Reply:
Yeah that's a good suggestion. Will do